packages/in_app_purchase/lib/src/store_kit_wrappers/sk_product_wrapper.dart

```diff
@@ -0,0 +1,179 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:flutter/foundation.dart';
+
+/// A unit discription the length of a subscription period.
+///
+/// This is wrapper of StoreKit's [SKProductPeriodUnit](https://developer.apple.com/documentation/storekit/skproductperiodunit?language=objc).
+enum SubscriptionPeriodUnit {
+  day,
+  week,
+  month,
+  year,
+}
+
+/// A subscription period.
+///
+/// A period is defined by a [numberOfUnits] and a [unit], e.g for a 3 months period [numberOfUnits] is 3 and [unit] is a month.
+///
+/// This is a wrapper for StoreKit's [SKProductSubscriptionPeriod](https://developer.apple.com/documentation/storekit/skproductsubscriptionperiod?language=objc).
+class SKProductSubscriptionPeriodWrapper {
+  SKProductSubscriptionPeriodWrapper(
+      {@required this.numberOfUnits, @required this.unit});
+
+  /// Constructor to build with a map.
+  ///
+  /// Used for constructing the class with the map passed from the OBJC layer.
+  SKProductSubscriptionPeriodWrapper.fromMap(Map<String, dynamic> map)
+      : numberOfUnits = map['numberOfUnits'],
+        unit = (map['unit'] != null)
+            ? SubscriptionPeriodUnit.values[map['unit']]
```

Comment:
Do these enum values match across both platforms? eg `SkProductPeriodUnit.day` is 0, `SkProductDiscountPaymentMode.PayAsYouGo` is 0, etc.

Reply:
Yes it matches the objc code.